File: pset1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float cash;
    
    //checks if cash is positive
    do
    {
        cash = get_float("Change owed: ");
    }
    while (cash < 0);
    
    //convert and rounds to an integer number
    int cents = round(cash * 100);
    
    int coins = 0;
    
    //check if it is possible to decrease
    while (cents >= 25)
    {
        cents = cents - 25;
        
        //count the number of change coins
        coins++;
    }
    
    while (cents >= 10)
    {
        cents = cents - 10;
        coins++;
    }
    
    while (cents >= 5)
    {
        cents = cents - 5;
        coins++;
    }
    
    while (cents >= 1)
    {
        cents = cents - 1;
        coins++;
    }
    
    printf("%i\n", coins);
}

File: pset6/cash.py
from cs50 import get_float

# Checks if cash is positive and re-prompt if it's not
while True:
    cash = get_float("Change owed: ")
    if cash > 0:
        break

# Convert and rounds to an integer number
cents = int(round(cash * 100))
coins = 0

# Check if it is possible to decrease in every case
while cents >= 25:
    cents -= 25
    # Count the number of change coins
    coins += 1

while cents >= 10:
    cents -= 10
    coins += 1
    
while cents >= 5:
    cents -= 5
    coins += 1

while cents >= 1:
    cents -= 1
    coins += 1

print(f"{coins}")s